edx_toggles/toggles/internal/waffles.py

```diff
@@ -0,0 +1,132 @@
+"""
+import logging
+
+from contextlib import contextmanager
+from waffle import switch_is_active
+
+from .base import BaseNamespace, BaseToggle
+
+log = logging.getLogger(__name__)
+
+
+class WaffleSwitchNamespace(BaseNamespace):
+    """
+    Provides a single namespace for a set of waffle switches.
+
+    All namespaced switch values are stored in a single request cache containing
+    all switches for all namespaces.
+    """
+
+    def is_enabled(self, switch_name):
+        """
+        Returns and caches whether the given waffle switch is enabled.
+        """
+        namespaced_switch_name = self._namespaced_name(switch_name)
+        value = self._cached_switches.get(namespaced_switch_name)
+        if value is None:
+            value = switch_is_active(namespaced_switch_name)
+            self._cached_switches[namespaced_switch_name] = value
+        return value
+
+    @contextmanager
+    def override(self, switch_name, active=True):
+        """
+        Overrides the active value for the given switch for the duration of this
+        contextmanager.
+        Note: The value is overridden in the request cache AND in the model.
+        """
+        previous_active = self.is_enabled(switch_name)
+        try:
+            self.override_for_request(switch_name, active)
+            with self.override_in_model(switch_name, active):
+                yield
+        finally:
+            self.override_for_request(switch_name, previous_active)
+
+    def override_for_request(self, switch_name, active=True):
+        """
+        Overrides the active value for the given switch for the remainder of
+        this request (as this is not a context manager).
+        Note: The value is overridden in the request cache, not in the model.
+        """
+        namespaced_switch_name = self._namespaced_name(switch_name)
+        self._cached_switches[namespaced_switch_name] = active
+        log.info(
+            "%sSwitch '%s' set to %s for request.",
+            self.log_prefix,
+            namespaced_switch_name,
+            active,
+        )
+
+    @contextmanager
+    def override_in_model(self, switch_name, active=True):
+        """
+        Overrides the active value for the given switch for the duration of this
+        contextmanager.
+        Note: The value is overridden in the model, not the request cache.
+        Note: This should probably be moved to a test class.
+        """
+        # Import is placed here to avoid model import at project startup.
+        # pylint: disable=import-outside-toplevel
+        from waffle.testutils import override_switch as waffle_override_switch
+
+        namespaced_switch_name = self._namespaced_name(switch_name)
+        with waffle_override_switch(namespaced_switch_name, active):
+            log.info(
+                "%sSwitch '%s' set to %s in model.",
+                self.log_prefix,
+                namespaced_switch_name,
+                active,
+            )
+            yield
```

Comment:
I'm hoping this can go away, but needs some investigation.  I feel like this was added at the beginning, and used with initial grades rollout (if I'm remembering correctly), but I don't think this is used in general and I'm not clear why all of this is needed.  I don't imagine most switches use this.

Reply:
Ok I'm adding this to the list of things to be investigated later.

Reply:
FYI: This was moved back to edx-platform.